zerver/lib/slack_data_to_zulip_data.py

```diff
@@ -300,13 +300,29 @@ def channel_message_to_zerver_message(constants: List[Any], channel: str,
 
     slack_usermention_regex = r"(<@)([a-zA-Z0-9]+)(\|)?([a-zA-Z0-9]+)?(>)"
 
-    # TODO mapping italics, bold, strikethrough etc
+    # Slack doesn't have mid-word message-formatting like Zulip.:w
+    slack_strikethrough_regex = r"(^|\"|\'|\(|\s)(~)([ -}]*)([!-}]+)(~)(\.|\)|\"\'|\s|$|\,)"
```

Comment:
Should there be test cases, maybe just as comments for what should be captured and shouldn't?

Reply:
![slack message formatting example.](https://user-images.githubusercontent.com/25330892/34581406-4928836a-f1b6-11e7-9718-3992056912c2.png)
I couldn't find a documentation on this, but this is the example in the picture. The message will be formatted only if it is the starting and the ending character, but only a few characters can be allowed before and after it like this characters in this list: `('.', ')', '(', '"', '''` and hence the regex.

Reply:
I'll be adding the test cases. Should I add it in `zerver/tests` with the name `test_slack_importer`?

Reply:
Hmmm, there are no test cases for regexes even in bugdown, but at least the lengthy one has a documentation for each part of the regex https://github.com/zulip/zulip/blob/master/zerver/lib/bugdown/__init__.py#L70-L75. Maybe just write an explanation of the intent the regex would suffice.

Reply:
A slack example messages-set json converted to an expected zulip messages-set json should already suffice if this does not already exists.

Reply:
The regex is a constant, so it should be `SLACK_STRIKETHROUGH_REGEX`.

Reply:
I have added the explanations for the regex. I have made sure to test every ascii char for this regex, (as only a few ascii chars can be allowed as the starting and ending chars, there is no documentation for this however, and adding a  message-set json for every char would get tedious. How should I go ahead?